### PR TITLE
[Feature] Initial Setup of Student Resource and Re-Ordering of Modules

### PR DESCRIPTION
**Pull request by Orrison**
Per our discussion earlier today @dgoetzit, I have rearranged the modules to fit our settled pattern.

I also setup the direct Model morph mapping for both the case and adm modules.

And finally created the resource for Student to continue work on it within future PRs.

### CODE REVIEW

File: app-modules/assist-data-model/routes/assist-data-model-module-routes.php
Comment (dgoetzit) [inline app-modules/assist-data-model/routes/assist-data-model-module-routes.php:1]: Not a big deal but we should remove the `module` from the name here.

File: composer.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "require": {
         "php": "^8.2",
         "ext-pdo": "*",
-        "canyon-gbs/case-module": "*",
+        "canyon-gbs/assist-data-model": "^1.0",
+        "canyon-gbs/case": "^1.0",
```
Comment (dgoetzit) [inline composer.json:16]: Should these be versioned at this point?

**Review by dgoetzit — APPROVED** [review]
Looks good. Just a couple of non-blocking questions/comments.

File: composer.json (hunk 1 above)
Comment (Orrison) [reply composer.json]: Ah no, that was a mistake with something I was doing.

**Review by dgoetzit — APPROVED** [review]

**Pull request merged**